pkg/query/build_tree.go

```diff
@@ -194,20 +194,20 @@ func (b *iteratorBuilder) buildIteratorFromOperation(p *schema.Permission, op sc
 
 		return NewExclusion(mainIt, excludedIt), nil
 
-	case *schema.FunctionedTuplesetOperation:
-		rel, ok := p.Parent().Relations()[perm.TuplesetRelation()]
+	case *schema.FunctionedArrowReference:
```

Comment:
It wasn't obvious to me that this is always supposed to be an arrow. I took it as a general function call, that happened to return arrows at the moment. Couldn't a functioned tupleset also be some sort of selection function, or multiple arrows (perhaps morphism style) or.... any number of things in the future?